src/common/utils/currencyConvert.ts

```diff
@@ -17,61 +15,11 @@ const getCurrencySettings = async () => {
 
 const numSatsInBtc = 100_000_000;
 
-const getFiatBtcRate = async (currency: CURRENCIES): Promise<string> => {
-  const { exchange } = await getCurrencySettings();
-
-  let response;
-
-  if (exchange === "yadio") {
-    response = await axios.get(
-      `https://api.yadio.io/exrates/${currency.toLowerCase()}`
-    );
-    const data = await response?.data;
-
-    return data.BTC;
-  }
-
-  if (exchange === "coindesk") {
-    response = await axios.get(
-      `https://api.coindesk.com/v1/bpi/currentprice/${currency.toLowerCase()}.json`
-    );
-    const data = await response?.data;
-
-    return data.bpi[currency].rate_float;
-  }
-
-  response = await axios.get(
-    `https://getalby.com/api/rates/${currency.toLowerCase()}.json`
-  );
-  const data = await response?.data;
-
-  return data[currency].rate_float;
-};
-
-// @TODO: https://github.com/getAlby/lightning-browser-extension/issues/1021
-// Replace decounce by saving rate to app-cache and only get it every minute for the whole app
-//
-// https://github.com/lodash/lodash/issues/4400#issuecomment-834800398
-const debouncedGetFiatBtcRate = debounce(
-  async function (callback) {
-    return await callback();
-  },
-  60000,
-  {
-    leading: true,
-    trailing: false,
-  }
-);
-
 const bitcoinToFiat = async (
   amountInBtc: number | string,
-  convertTo: CURRENCIES,
-  isLatestRate?: boolean
+  convertTo: CURRENCIES
 ) => {
-  const rate = isLatestRate
-    ? await getFiatBtcRate(convertTo)
-    : await debouncedGetFiatBtcRate(() => getFiatBtcRate(convertTo));
-
+  const { rate } = await getCurrencyRate({ currency: convertTo });
```

Comment:
is this call is done for every amount that we want to convert?

Reply:
Yes, is there an issue with this? What are you worried about?

Reply:
it's sending some message, doing some storage lookups, etc. then for every amount that we want to display on the page. 
Even though it might work I am not sure if this is a good idea? 

we load the settings on each page load, correct? can't we include the current rate for the configured currency in that settings call and use it from there? - just thinking out loud.